src/components/Articles.jsx

```diff
@@ -1,32 +1,39 @@
 import { useEffect, useState } from 'react';
-import { useParams } from 'react-router-dom';
+import { Link, useLocation } from 'react-router-dom';
 import { getArticles } from '../utils/api';
+import '../css/Articles.css';
 
 const Articles = () => {
 	const [articles, setArticles] = useState([]);
-	const { topic } = useParams();
 
+	const queries = new URLSearchParams(useLocation().search);
+	const sortBy = queries.get('sort_by');
+	const order = queries.get('order');
+	const topic = queries.get('topic');
 	useEffect(() => {
-		getArticles(topic)
+		getArticles(sortBy, order, topic)
 			.then((articlesFromApi) => {
 				setArticles(articlesFromApi);
 			})
 			.catch((err) => console.log(err));
-	}, [topic]);
+	}, [sortBy, order, topic]);
+
+	const pageTitle = topic ? `Articles on ${topic}` : 'All Articles';
 
 	if (articles.length === 0) {
 		return <h2>Loading...</h2>;
 	} else {
 		return (
 			<div>
-				<h2>Articles</h2>
-				<ul>
+				<h2>{pageTitle}</h2>
+				<ul className='Articles__article-list'>
 					{articles.map((article) => {
 						const { article_id, title, author, votes, comment_count } = article;
 						return (
 							<li key={article_id}>
-								<h3>{title}</h3>
-
+								<h3>
```

Comment:
We could refactor this at some point and extract this out into its own component. The benefits of this are that it will make our code look neater, and allow to reuse that component anywhere within our application, should we wish to do so.

Reply:
Good point, added that in, thanks